sklearn/utils/estimator_checks.py

```diff
@@ -2359,6 +2362,89 @@ def check_classifiers_regression_target(name, estimator_orig):
         assert_raises_regex(ValueError, msg, e.fit, X, y)
 
 
+@ignore_warnings(category=(DeprecationWarning, FutureWarning))
+def check_estimator_sparse_dense(name, estimator_orig):
+    rng = np.random.RandomState(52)
```

Comment:
Finding the right seed to make the test pass doesn't feel right at all :/

Reply:
I agree :p that's what we did with @agramfort and after discussion with @ogrisel because GradientBoosting worked but some times and some times not. But that's also why I commented https://github.com/scikit-learn/scikit-learn/pull/13246#discussion_r260758475 and added this to the todo: 

```
- [ ] Try to be more robust by maybe testing only the score (if possible), with a certain tolerance, and for certain random seeds (or if testing the predict also maybe test that like 90% of predictions are the same ?)
```

Does this sounds a good strategy ?

(Testing for a lot of random seeds, but with a certain tolerance. And for predictions since they are hard values test something like 95% of the values are the same or sth like that)

Reply:
Eventually I put a friendly dataset as discussed with @gramfort, so now it solves the problem

Reply:
This is bad to have checks that depend on a particular seed. Those checks are not meant to be used only by scikit-learn but also by third party libraries. So we need to make sure that the check is generic enough to avoid brittle failures (unless its caused by a limitation of a particular estimator in which case it should be XFAILED with an estimator tag and a proper motivation).

> Eventually I put a friendly dataset as discussed with @gramfort, so now it solves the problem

So this mean that this check would pass for all scikit-learn estimators with a canonical random seed such as 0 or 42? If so please change the seed to 0 or 42 because 52 makes us think that it's a special seed.